test/sanitytest.c

```diff
@@ -76,12 +76,20 @@ static int test_sanity_unsigned_convertion(void)
     return 1;
 }
 
+static int test_sanity_pointer_sizes(void)
+{
+    if (!TEST_int_eq(sizeof(long), sizeof(void *)))
+        return 0;
+    return 1;
+}
+
 void register_tests(void)
 {
     ADD_TEST(test_sanity_null_zero);
     ADD_TEST(test_sanity_enum_size);
     ADD_TEST(test_sanity_twos_complement);
     ADD_TEST(test_sanity_sign);
     ADD_TEST(test_sanity_unsigned_convertion);
+    ADD_TEST(test_sanity_pointer_sizes);
```

Comment:
If you mean that non-equality would mean test fail, then it's wrong assumption. As there is *supported* platform with sizeof(long) != sizeof(void *), 64-bit Windows.